lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/MultiRuntimeYarnBootstrapDriverConfigGenerator.java

```diff
@@ -0,0 +1,310 @@
+import org.apache.reef.wake.remote.ports.parameters.TcpPortRangeTryCount;
+
+import javax.inject.Inject;
+import java.io.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This is the Java Driver configuration generator for .NET Drivers that generates
+ * the Driver configuration at runtime for multiruntime. Called by {@link MultiRuntimeYarnBootstrapREEFLauncher}.
+ */
+final class MultiRuntimeYarnBootstrapDriverConfigGenerator {
+  private static final Logger LOG = Logger.getLogger(MultiRuntimeYarnBootstrapDriverConfigGenerator.class.getName());
+  private static final String DUMMY_YARN_RUNTIME = "DummyYarnRuntime";
+
+  private final MultiRuntimeDefinitionSerializer runtimeDefinitionSerializer = new MultiRuntimeDefinitionSerializer();
+
+  private final REEFFileNames reefFileNames;
+  private final ConfigurationSerializer configurationSerializer;
+
+  @Inject
+  private MultiRuntimeYarnBootstrapDriverConfigGenerator(final REEFFileNames reefFileNames,
+                                                         final ConfigurationSerializer configurationSerializer) {
+    this.configurationSerializer = configurationSerializer;
+    this.reefFileNames = reefFileNames;
+  }
+
+  /**
+   * Adds yarn runtime definitions to the builder.
+   * @param yarnJobSubmissionParams Yarn job submission parameters
+   * @param jobSubmissionParameters Generic job submission parameters
+   * @param builder The multi runtime builder
+   */
+  private void addYarnRuntimeDefinition(
+          final AvroYarnJobSubmissionParameters yarnJobSubmissionParams,
+          final AvroJobSubmissionParameters jobSubmissionParameters,
+          final MultiRuntimeDefinitionBuilder builder) {
+    // create and serialize yarn configuration if defined
+    final Configuration yarnDriverConfiguration =
+            createYarnConfiguration(yarnJobSubmissionParams, jobSubmissionParameters);
+
+    // add yarn runtime to the builder
+    builder.addRuntime(yarnDriverConfiguration, RuntimeIdentifier.RUNTIME_NAME);
+  }
+
+  private Configuration createYarnConfiguration(
+          final AvroYarnJobSubmissionParameters yarnJobSubmissionParams,
+          final AvroJobSubmissionParameters jobSubmissionParameters) {
+    return YarnDriverConfiguration.CONF
+            .set(YarnDriverConfiguration.JOB_SUBMISSION_DIRECTORY,
+                    yarnJobSubmissionParams.getDfsJobSubmissionFolder().toString())
+            .set(YarnDriverConfiguration.JOB_IDENTIFIER,
+                    jobSubmissionParameters.getJobId().toString())
+            .set(YarnDriverConfiguration.CLIENT_REMOTE_IDENTIFIER,
+                    ClientRemoteIdentifier.NONE)
+            .set(YarnDriverConfiguration.JVM_HEAP_SLACK, 0.0)
+            .set(YarnDriverConfiguration.RUNTIME_NAMES, RuntimeIdentifier.RUNTIME_NAME)
+            .build();
+  }
+
+  /**
+   * Adds yarn runtime definitions to the builder, with a dummy name, to prevent evaluator submission.
```

Comment:
Will add docs. This is needed to enforce usage of specified runtimes, so if user did not specify yarn runtime for evaluators we still need to instantiate one (as it registers with RM) but won't allow submitting evaluators to it.